File: Backend/src/Book.cpp
#include "Book.h"

Book::Book()
    : title{""}, author{""}, isbn{"000-0-00-000000-0"},
      borrowed{false}, borrowed_to{""}, borrowed_from{""}
{
}

Book::Book(std::string title,
           std::string author,
           Isbn isbn)
{
    this->title = title;
    this->author = author;
    this->isbn = isbn;
    this->borrowed = false;
    // how to assign properly?
    this->borrowed_to{""};
    this->borrowed_from{""};
}

File: Backend/DB_conn/src/db_conn/Book.java
package db_conn;

public class Book {
    private String title;
    private Person[] authors;
    private Isbn isbn;

    private boolean borrowed;
    private Person borrowed_by;

    public Book() {
        this.title = "";
        this.authors = new Person[0];
        this.isbn = new Isbn();
        borrowed = false;
        borrowed_by = new Person();
    }

    public Book(String title, Person[] authors, Isbn isbn) {
        this.title = title;
        this.authors = authors;
        this.isbn = isbn;
        borrowed = false;
        borrowed_by = new Person();
    }

    public String getTitle() {
        return this.title;
    }

    public Person[] getAuthors() {
        return this.authors;
    }

    public Isbn getIsbn() {
        return this.isbn;
    }

    public Boolean getBorrowed() {
        return this.borrowed;
    }

    public Person getBorrower() {
        return this.borrowed_by;
    }

    @Override
    public String toString() {
        // arbitrary
        return this.title + "; " + this.authors.toString() + "; " + this.isbn.getIsbn();
    }
}
